From 625b53f516d27eb61eaad00175dd7108bf82572f Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Tue, 16 Aug 2022 00:12:00 +0000
Subject: [PATCH] Don't pass Exec arguments to kstart when launching via
 desktop entry name

If we have a desktop file org.kde.foo.desktop with "Exec=foo bar bla"

we are currently calling "kstart5 --application org.kde.foo bar bla"

However we need to call "kstart5 --application org.kde.foo" without any args

kstart5/ApplicationLauncherJob will read the Exec line again and pick up the args

BUG: 440507
FIXED-IN: 5.98
---
 src/runtime/kserviceactioncomponent.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/runtime/kserviceactioncomponent.cpp b/src/runtime/kserviceactioncomponent.cpp
index 48e1fe3..c7ea5c7 100644
--- a/src/runtime/kserviceactioncomponent.cpp
+++ b/src/runtime/kserviceactioncomponent.cpp
@@ -77,13 +77,12 @@ void KServiceActionComponent::runProcess(const KConfigGroup &group, const QStrin
     const auto kstart = QStandardPaths::findExecutable(QStringLiteral("kstart5"));
     if (!kstart.isEmpty()) {
         if (group.name() == QLatin1String("Desktop Entry") && m_isInApplicationsDir) {
-            args.prepend(QFileInfo(m_desktopFile->fileName()).completeBaseName());
-            args.prepend(QStringLiteral("--application"));
+            startDetachedWithToken(kstart, {QStringLiteral("--application"), QFileInfo(m_desktopFile->fileName()).completeBaseName()});
         } else {
             args.prepend(command);
             args.prepend(QStringLiteral("--"));
+            startDetachedWithToken(kstart, args);
         }
-        startDetachedWithToken(kstart, args);
         return;
     }
 
-- 
GitLab

